src/lib/components/Modal/README.mdx

```diff
@@ -239,19 +488,119 @@ Modal can be aligned either to top or center of the screen.
   }}
 </Playground>
 
-## Scrolling Long Content
+## Key Control
+
+Modal can be controlled either by mouse or keyboard. To enhance user
+experience, primary action can be fired by pressing `Enter` key and modal
+can be closed by pressing `Escape` key.
+
+To enable it, you just need to pass reference to the buttons using
+`primaryButtonRef` and `modalCloseButtonRef` props on Modal. Advantage
+of passing reference to the button is that if the button is disabled,
+key press will not fire the event.
+
+👉 We strongly recommend using this feature together with Autofocus for
+better user experience.
+
+## Autofocus
 
-When modals become too long for the user's viewport or device, they scroll
-independent of the page itself.
+Autofucus is implemented to enhance user experience by automatically
+focussing an element within the modal.
 
-The inner implementation of body scrolling uses a partially
-[customizable](#api) instance of the [ScrollView](/components/scroll-view)
-component.
+How does it work? It tries to find `input`, `textarea` and `select` element
+inside of Modal and focuses the first found. If none is found and
+`primaryButtonRef` prop on Modal is set, then the primary button is focused.
+
+Autofocus is enabled by default, so if you want to control focus of
+elements manually, set `autoFocus` prop on Modal to `false`.
+
+## Scrolling Long Content
+
+When modals become too long for the user's viewport or device, we recommend
+wrapping ModalContent into [ScrollView](/components/scroll-view) to make
+its content scrollable independently of the page itself. In this case,
+`withScrollView` must be set to `true` on ModalBody.
 
 <Playground>
   {() => {
     const [modalOpen, setModalOpen] = React.useState(false);
     const [modalScrollView, setModalScrollView] = React.useState(true);
+    const modalCloseButtonRef = React.useRef();
+    const modalPrimaryButtonRef = React.useRef();
+    const modalContent = (
```

Comment:
I would consider putting all content into one looong line. This makes reading the code (both in IDE and in the live preview) a pain.

Reply:
We can't due to linter restrictions, tests would fail. And the second thing is that is untouched part written originally by @adamkudrna, I just split Modal into multiple components.

Reply:
> We can't due to linter restrictions

I was assuming there was some way to include an inline ignore directive. There is apparently not: https://github.com/markdownlint/markdownlint/issues/16

Lets don't worry about it then